### PR TITLE
[Enhancement](Load) Nereids supports http_stream and group_commit with stream load

### PR DESCRIPTION
**Pull request by zzzzzzzs**
This PR is from https://github.com/apache/doris/pull/25049. Due to https://github.com/apache/doris/pull/25049 not merge for a long time, there was a problem during rebase, so a new PR was created.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzzzzzzs** [comment]
run buildall

**Comment by zzzzzzzs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -418,12 +423,26 @@ public PlanFragment visitPhysicalOlapTableSink(PhysicalOlapTableSink<? extends P
             slotDesc.setIsNullable(column.isAllowNull());
             slotDesc.setAutoInc(column.isAutoInc());
         }
-        OlapTableSink sink = new OlapTableSink(
+        checkInnerGroupCommit(context.getScanNodes());
+        OlapTableSink sink;
+        if (context.getConnectContext().isGroupCommitTvf()) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:428]: should not implement and logic code in translator, translator should only transalte Nereids' plan to legacy one

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/InsertIntoTableCommand.java [hunk 2]
```diff
@@ -190,13 +194,18 @@ private void runInternal(ConnectContext ctx, StmtExecutor executor) throws Excep
...
         insertExecutor.getCoordinator().getQueryOptions().setEnableMemtableOnSinkNode(isEnableMemtableOnSinkNode);
         executor.setProfileType(ProfileType.LOAD);
         // We exposed @StmtExecutor#cancel as a unified entry point for statement interruption
         // so we need to set this here
         executor.setCoord(insertExecutor.getCoordinator());
+        return insertExecutor;
+    }
+
+    private void runInternal(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        InsertExecutor insertExecutor = initPlan(ctx, executor);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/InsertIntoTableCommand.java:208]:
```suggestion
        initPlan(ctx, executor);
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 3]
```diff
@@ -1782,10 +1795,12 @@ public PlanFragment visitPhysicalProject(PhysicalProject<? extends Plan> project
             if (requiredByProjectSlotIdSet.size() != requiredSlotIdSet.size()
                     || new HashSet<>(projectionExprs).size() != projectionExprs.size()
                     || projectionExprs.stream().anyMatch(expr -> !(expr instanceof SlotRef))) {
-                projectionTuple = generateTupleDesc(slots,
+                if (!(inputPlanNode instanceof GroupCommitScanNode)) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:1798]: add comment to explain why need this if statement

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 4]
```diff
@@ -410,12 +415,20 @@ public PlanFragment visitPhysicalOlapTableSink(PhysicalOlapTableSink<? extends P
             slotDesc.setIsNullable(column.isAllowNull());
             slotDesc.setAutoInc(column.isAutoInc());
         }
-        OlapTableSink sink = new OlapTableSink(
+        checkInnerGroupCommit(context.getScanNodes());
+        OlapTableSink sink;
+        if (context.getConnectContext().isGroupCommitStreamLoadSql()) {
+            sink = new GroupCommitBlockSink(olapTableSink.getTargetTable(), olapTuple,
+                olapTableSink.getTargetTable().getPartitionIds(), olapTableSink.isSingleReplicaLoad(),
+                ConnectContext.get().getSessionVariable().getGroupCommit(), 0);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:423]: add comment to explain why need this if statement

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 5]
```diff
@@ -2607,4 +2622,13 @@ private List<Expr> getDistributeExpr(List<ExprId> childOutputIds, DistributionSp
         }
         return Lists.newArrayList();
     }
+
+    private void checkInnerGroupCommit(List<ScanNode> scanNodes) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:2626]: add comment to explain the diff between GroupCommitTableValuedFunction and other tvf

**Pull request closed without merging**